File: Algorithms/EuclideGCD.py
def findGCD(a, b):
    if (a == 0):
        return b
    elif (b == 0):
        return a
    elif (a >= b):
        return findGCD(a % b, b)
    elif (b >= a):
        return findGCD(a, b % a)


def main():
    a, b = map(int, input().split())
    print(findGCD(a, b))


if __name__ == "__main__":
    main()

File: Algorithms/EuclideAlgorithm/EuclideGCD.cs
using System.Diagnostics;


namespace EuclideGCD
{
    public static class EuclideGCD
    {
        public static (int, int) ParseInput(this string input)
        {
            string[] chars = input.Split(' ');
            return (int.Parse(chars[0]), int.Parse(chars[1]));
        }

        public static int GetGCD(int number1, int number2)
        {
            Debug.Assert(number1 > 0);

            if (number2 > 0)
            {
                return GetGCD(number2, number1 % number2);
            }

            return number1;
        }

        public static void Main()
        {
            string? input = Console.ReadLine();
            (int number1, int number2) = input.ParseInput();
            Console.WriteLine(GetGCD(number1, number2));
        }
    }
}